crates/biome_js_type_info/src/inference.rs

```diff
@@ -173,6 +145,125 @@ impl Type {
         }
     }
 
+    pub fn from_any_js_expression(expr: &AnyJsExpression) -> Self {
```

Comment:
This method returns `Self` while `from_any_js_literal_expression` returns `Option<Self>`. Should they return the same? Or should we comment on the one that is the "block sheep"?

Reply:
It's really just a matter of convenience. I try to use `Self` unless I have a good reason otherwise. With `from_any_js_literal_expression`, the reason was simply that I could use `?` :sweat_smile: